### PR TITLE
Improvements for kubect-minio plugin

### PR DESCRIPTION
**Pull request by Alevsk**
- Use secure GenerateCredentials() function for tenants created via the
  krew plugin
- Use configuration secret instead of credsSecret for tenants created
  via the krew plugin

Signed-off-by: Lenin Alevski <alevsk.8772@gmail.com>

### CODE REVIEW

**Review by dvaldivia — DISMISSED** [review]

File: pkg/apis/minio.min.io/v2/utils.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		"/", "+")
+
+	return accessKey, secretKey, nil
+}
+
+// GenerateTenantConfigurationFile :
+func GenerateTenantConfigurationFile(configuration map[string]string) string {
+	var rawConfiguration string
+	for key, val := range configuration {
+		rawConfiguration += fmt.Sprintf("export %s=\"%s\"\n", key, val)
```
Comment (harshavardhana) [inline pkg/apis/minio.min.io/v2/utils.go:80]:
```suggestion
		rawConfiguration += fmt.Sprintf(`export %s="%s"\n`, key, val)
```

Use quote style to simplify escaping @Alevsk

File: pkg/apis/minio.min.io/v2/utils.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+	secretKey = strings.ReplaceAll(string([]byte(base64.StdEncoding.EncodeToString(keyBytes))[:secretKeyMaxLen]),
+		"/", "+")
+
+	return accessKey, secretKey, nil
+}
+
+// GenerateTenantConfigurationFile :
+func GenerateTenantConfigurationFile(configuration map[string]string) string {
+	var rawConfiguration string
```
Comment (harshavardhana) [inline pkg/apis/minio.min.io/v2/utils.go:78]:
```suggestion
	var rawConfiguration strings.Builder
```

File: pkg/apis/minio.min.io/v2/utils.go (hunk 1 above)
Comment (harshavardhana) [reply pkg/apis/minio.min.io/v2/utils.go:80]:
```suggestion
		rawConfiguration.WriteString(fmt.Sprintf(`export %s="%s"\n`, key, val))
```

File: pkg/apis/minio.min.io/v2/utils.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	for key, val := range configuration {
+		rawConfiguration += fmt.Sprintf("export %s=\"%s\"\n", key, val)
+	}
+	return rawConfiguration
```
Comment (harshavardhana) [inline pkg/apis/minio.min.io/v2/utils.go:82]:
```suggestion
	return rawConfiguration.String()
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: kubectl-minio/cmd/resources/secret.go [hunk 4]
```diff
@@ -16,52 +16,73 @@
...
 		},
 		TypeMeta: metav1.TypeMeta{
 			Kind:       "Secret",
 			APIVersion: v1.SchemeGroupVersion.Version,
 		},
-		Data: consoleSecretData(),
-	}
+		Data: map[string][]byte{
+			"CONSOLE_ACCESS_KEY": []byte(accessKey),
+			"CONSOLE_SECRET_KEY": []byte(secretKey),
```
Comment (harshavardhana) [inline kubectl-minio/cmd/resources/secret.go:85]: is this needed?  there is no such thing as `CONSOLE_ACCESS_KEY` and `CONSOLE_SECRET_KEY` anymore..

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: kubectl-minio/cmd/resources/secret.go (hunk 4 above)
Comment (Alevsk) [reply kubectl-minio/cmd/resources/secret.go:85]: those entries are used by the operator to create initial users for a tenant (`func (t *Tenant) CreateUsers` function), we can rename them at some point to just `access_key` and `secret_key`, this includes the `console_` prefix for historical reasons.

Comment (harshavardhana) [reply kubectl-minio/cmd/resources/secret.go:85]: okay, I see perhaps let's leave it.

**Review by harshavardhana — APPROVED** [review]

**Review by dvaldivia — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**